src/js/interface.js

```diff
@@ -447,6 +447,9 @@ window.addEventListener("load", function() {
     }
 
     function convertOutput(value) {
+        if( convertOutput.highSurrogate === undefined ) {
+            convertOutput.highSurrogate = 0;
```

Comment:
I am curious: how would you describe this static variable to students? Would you call this a register in the output peripheral, which holds the high surrogate value?

Reply:
No, as far as students of MARIE are concerned, I would simply say that interpretation of output is completely separate from generating it, otherwise the drop-down for the output box wouldn't even make sense. Some output formats (most notably of course all the UTF variable-length encodings) require a minimal amount of state to generate a certain output token. This may be implemented in software, it may be implemented in hardware via a flag and/or a register, I think this is ultimately up to the output interface designers. All we promise them is a UTF-16 compatible output stream.